### PR TITLE
tests: add integration test

### PR DESCRIPTION
**Pull request by mrcjkb**
For now, I have set `--option sandbox false`, to work around https://github.com/luarocks/luarocks/issues/1539 (which is not ideal).

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+trim_trailing_whitespace = true
+
+[*.nix]
+indent_size = 2
+
+# Ignore paths
+[lua/nio/**]
+charset = unset
+end_of_line = unset
+insert_final_newline = unset
```
Comment (teto) [inline .editorconfig:17]: I am not familiar with the `unset`. What's the advantage of defining those ?

File: Makefile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 format:
-	stylua -v --verify lua/rocks/ plugin/
+    stylua -v --verify lua/rocks/ plugin/
```
Comment (teto) [inline Makefile:2]: seems like a mistake

File: .editorconfig (hunk 1 above)
Comment (mrcjkb) [reply .editorconfig:17]: editorconfig doesn't have an "ignore" config.
This is a workaround for that.

File: Makefile (hunk 2 above)
Comment (mrcjkb) [reply Makefile:2]: I think that was editorconfig complaining

Comment (teto) [reply Makefile:2]: I mean dont makefiles only accept tab before commands ?

**Review by teto — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (mrcjkb) [reply Makefile:2]: You're probably right ..

File: .luacheckrc [hunk 3]
```diff
@@ -1,6 +1,10 @@
 globals = {
     "_",
-    "vim"
+    "vim",
+    "describe",
+    "it",
+    "assert",
+    "stup",
```
Comment (NTBBloodbath) [inline .luacheckrc:7]: I'm not sure why this was added to the globals, maybe it was a typo? I couldn't find anything with that name in the code

**Review by NTBBloodbath — APPROVED** [review]
LGTM

**Pull request merged**